addons/website_slides/controllers/main.py

```diff
@@ -729,6 +731,25 @@ def slide_view(self, slide, **kwargs):
         values.pop('channel', None)
         return request.render("website_slides.slide_main", values)
 
+    @http.route('/slides/slide/<int:slide_id>/share', type='http', auth="public", website=True, sitemap=False)
+    def slide_shared_view(self, slide_id, **kwargs):
```

Comment:
We could handle the 'is_preview' flag here directly and redirect to the slide, instead of having to do it at field level. Seems better to let controller handle redirection and field handle the link computation.